Code/Framework/AzCore/Platform/Common/UnixLike/AzCore/Module/DynamicModuleHandle_UnixLike.cpp

```diff
@@ -72,13 +72,32 @@ namespace AZ
             {
                 if (auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
                 {
+                    bool fileFound = false;
                     if (AZ::IO::FixedMaxPath projectModulePath;
                         settingsRegistry->Get(projectModulePath.Native(), AZ::SettingsRegistryMergeUtils::FilePathKey_ProjectConfigurationBinPath))
                     {
                         projectModulePath /= fullFilePath;
                         if (AZ::IO::SystemFile::Exists(projectModulePath.c_str()))
                         {
                             m_fileName.assign(projectModulePath.c_str(), projectModulePath.Native().size());
+                            fileFound = true;
+                        }
+                    }
+                    
+                    if (!fileFound)
+                    {
+                        if (AZ::IO::FixedMaxPath installedBinariesPath;
+                            settingsRegistry->Get(installedBinariesPath.Native(), AZ::SettingsRegistryMergeUtils::FilePathKey_InstalledBinaryFolder))
+                        {
+                            if (AZ::IO::FixedMaxPath engineRootFolder;
+                                settingsRegistry->Get(engineRootFolder.Native(), AZ::SettingsRegistryMergeUtils::FilePathKey_EngineRootFolder))
+                            {
+                                installedBinariesPath = engineRootFolder / installedBinariesPath / fullFilePath;
+                                if (AZ::IO::SystemFile::Exists(installedBinariesPath.c_str()))
+                                {
+                                    m_fileName.assign(installedBinariesPath.c_str(), installedBinariesPath.Native().size());
+                                }
+                            }
```

Comment:
Wouldn't Windows need this same logic as well to read the InstalledBinaryFolder key as well?

Reply:
In Windows, I assumed if the dll was in the same directory as the executable, it would find it. We're having to deal with this here because the dylib may not always be inside the bundle.

Reply:
Since the dependencies are copied to the project configuration bin path when the user builds the project, and we check that location for the lib just above this block, I don't think this is needed for Mac either. The user will need to build their project before launching the AP/Editor anyway. I'm reverting this change.

Reply:
I just built AtomSampleViewer with the SDK and realized I'm going to need to add this back and the install path setreg in the AssetProcessor and Editor as well. The runtime dependencies are being copied into the GameLauncher bundle(which is in the project build directory i.e., <project_build_dir>/bin/<config>/<Game>.app/Contents/MacOS/lib<O3DE_GemDependency>.dylib) of the game project and not the project build directory which is the path in the settings registry(i.e., <project_build_dir>/bin/<config>). So we still need AP/Editor to have that setreg file so that it can look it up. But the setreg file doesn't need to be target specific. So I can use the same one that's being used by the `O3DE_SDK_InstallPath.o3de_sdk.setreg`. I'll just remove the specialization and rename it to `BinariesInstallPath.setreg`

Reply:
So this is needed for the AssetProcessor/Editor/O3DE_SDK to locate any "loose" gems within the installed binary directory that is outside of the bundle of those applications?

Reply:
Yes. The loose gems are dependencies of the game, not tool, and so were not included in the tool bundle when it was generated.